packages/main/src/plugin/util/exec.ts

```diff
@@ -20,16 +20,29 @@ import type { ChildProcessWithoutNullStreams } from 'child_process';
 import { spawn } from 'child_process';
 import type { RunError, RunOptions, RunResult } from '@podman-desktop/api';
 import { isMac, isWindows } from '../../util.js';
+import type { Proxy } from '../proxy.js';
 
 export const macosExtraPath = '/usr/local/bin:/opt/homebrew/bin:/opt/local/bin:/opt/podman/bin';
 
-export function exec(command: string, args?: string[], options?: RunOptions): Promise<RunResult> {
+export function exec(command: string, proxy: Proxy, args?: string[], options?: RunOptions): Promise<RunResult> {
```

Comment:
When I was wondering if proxy was placed at the right order it let me think that proxy is a global parameter and not a 'per call parameter'

so I think it might be better to replace the utility function exec by a constructor/class/object

like somewhere in plugins:

```
const executor = new Executor(proxy);
```

and when we invoke we just use
```
executor.exec(command, args, options);
```

so we don't need to pass Proxy to contribution manager and extension loader, but we give a reference to the exec handler
Only Exec has a direct dependency to proxy